File: frontend/src/utils/document-status.ts
import type { DocumentStatus } from '../api/contracts';

const inFlightStatuses: ReadonlySet<DocumentStatus> = new Set([
  'UPLOADED',
  'QUEUED',
  'PROCESSING',
  'RETRYING',
]);

export function isDocumentProcessing(status: DocumentStatus): boolean {
  return inFlightStatuses.has(status);
}

export function isDocumentTerminal(status: DocumentStatus): boolean {
  return status === 'READY' || status === 'FAILED' || status === 'DELETED';
}

export function hasInFlightDocuments(statuses: DocumentStatus[]): boolean {
  return statuses.some(isDocumentProcessing);
}

File: src/main/java/com/wxx/aidocumentagent/document/domain/DocumentStatus.java
package com.wxx.aidocumentagent.document.domain;

/**
 * 已上传文档的生命周期状态。
 */
public enum DocumentStatus {
    UPLOADED,
    QUEUED,
    PROCESSING,
    RETRYING,
    READY,
    FAILED,
    DELETED
}
